src/v/cluster/topic_table.cc

```diff
@@ -277,6 +277,37 @@ topic_table::apply(update_topic_properties_cmd cmd, model::offset o) {
     co_return make_error_code(errc::success);
 }
 
+ss::future<std::error_code>
+topic_table::apply(create_materialized_topic_cmd cmd, model::offset o) {
+    const model::topic_namespace& source = cmd.value.source;
+    const model::topic_namespace& new_materialized_topic
+      = cmd.value.materialized;
+    if (_topics.contains(new_materialized_topic)) {
+        co_return make_error_code(errc::topic_already_exists);
+    }
+    auto tp = _topics.find(source);
+    if (tp == _topics.end()) {
+        /// Source topic must exist if attempting to create a materialized topic
+        co_return make_error_code(errc::missing_materialized_source_topic);
+    }
+
+    for (const auto& pas : tp->second.configuration.assignments) {
```

Comment:
Assignments are consistent and deterministic, we can use an assignment from source partition as a source for assignment for materialized topic. The it will not have to stored with materialized topic configuration.